build-tools/automation/build.groovy

```diff
@@ -85,15 +133,19 @@ timestamps {
 
         stageWithTimeout('build', 6, 'HOURS', XADir, true) {    // Typically takes less than one hour except a build on a new bot to populate local caches can take several hours
             if (isPr) {
-                echo "PR build definition detected: building with 'make all'"
-                sh "make all CONFIGURATION=${env.BuildFlavor} MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                echo "PR build definition detected: building with 'make ${buildTarget}'"
+                sh "make prepare ${buildTarget} CONFIGURATION=${env.BuildFlavor} MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
             } else {
-                echo "PR build definition *not* detected: building with 'make jenkins'"
-                sh "make jenkins CONFIGURATION=${env.BuildFlavor} MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
+                echo "PR build definition *not* detected: building with 'make ${buildTarget}'"
+                sh "make ${buildTarget} CONFIGURATION=${env.BuildFlavor} MSBUILD_ARGS='$MSBUILD_AUTOPROVISION_ARGS'"
             }
         }
 
         stageWithTimeout('create vsix', 30, 'MINUTES', XADir, true) {    // Typically takes less than 5 minutes
+            if (isPr) {
+                sh "make package-oss"
```

Comment:
This should be moved out into it's on separate stage.

It'll be a no-op on `make jenkins` builds, as the output file will already exist.

Reply:
@jonpryor - so I should lose the isPr check?

Reply:
Created a new 'package oss' stage and removed the isPr check. commit https://github.com/xamarin/xamarin-android/pull/2767/commits/f1861c8a6ed6779a1ef790886ea1912232935a80